pkg/build/buildkit/client.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buildkit
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/url"
+	"os"
+
+	"github.com/moby/buildkit/client"
+	"github.com/okteto/okteto/pkg/log/io"
+	"golang.org/x/oauth2"
+	"google.golang.org/grpc/credentials/oauth"
+)
+
+const (
+	// readWritePermission is the permission for reading and writing
+	readWritePermission = 0600
+)
+
+// urlParser defines an interface for parsing URLs
+type urlParser interface {
+	Parse(rawurl string) (*url.URL, error)
+}
+
+// realURLParser implements the urlParser interface
+type realURLParser struct{}
+
+// Parse parses a URL
+func (p *realURLParser) Parse(rawurl string) (*url.URL, error) {
+	return url.Parse(rawurl)
+}
+
+// certDecoder defines an interface for decoding certificates
+type certDecoder interface {
+	DecodeString(s string) ([]byte, error)
+}
+
+// base64CertDecoder implements the certDecoder interface
+type base64CertDecoder struct{}
+
+// DecodeString decodes a base64 string
+func (b *base64CertDecoder) DecodeString(s string) ([]byte, error) {
+	return base64.StdEncoding.DecodeString(s)
+}
+
+type fileWriter interface {
+	WriteFile(filename string, data []byte, perm os.FileMode) error
+}
+
+// osFileWriter implements the fileWriter interface
+type osFileWriter struct{}
+
+// WriteFile writes data to a file
+func (o *osFileWriter) WriteFile(filename string, data []byte, perm os.FileMode) error {
+	return os.WriteFile(filename, data, perm)
+}
+
+// buildkitClientCreator defines an interface for creating buildkit clients
+type buildkitClientCreator interface {
```

Comment:
```suggestion
type clientCreator interface {
```

Nit. This is already within a package called `buildkit`. Would it make sense to just call it `clientCreator`?